text/trace/0157-messaging-semantic-conventions.md

```diff
@@ -0,0 +1,184 @@
+state. The goal is to declare messaging semantic conventions stable before the
+end of 2021.
+
+## Motivation
+
+Many observability scenarios involve messaging systems. For Distributed Tracing
+to be useful across the entire scenario, having good observability for
+messaging operations is critical. To achieve this, OpenTelemetry must provide
+stable conventions and guidelines for instrumenting messaging systems.
+
+Bringing the existing experimental semantic conventions for messaging to a
+stable state is a crucial step for users and instrumentation authors, as it
+allows them to rely on [stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability),
+and thus to ship and use stable instrumentation.
+
+## Roadmap
+
+| Description | Done By     |
+|-------------|-------------|
+| This OTEP, consisting of scenarios and a proposed roadmap, is approved and merged. | 09/30/2021 |
+| [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability) for semantic conventions are approved and merged. This is not strictly related to semantic conventions for messaging but is a prerequisite for stabilizing any semantic conventions. | 09/30/2021 |
+| An OTEP proposing a set of attributes and conventions covering the scenarios in this document is approved and merged. | 10/29/2021 |
+| Proposed specification changes are verified by prototypes for the scenarios and examples below. | 11/15/2021 |
+| The [specification for messaging semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md) are updated according to the OTEP mentioned above and are declared [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable). | 11/30/2021 |
+
+## Terminology
+
+### Message
+
+A "message" is a transport wrapper for the transfer of information between two
+or more parties. The information is a combination of data and metadata.
+Messages may be transferred directly between parties or via one or more
+intermediaries.  A message may carry annotations that are directed at
+intermediaries on the message path. Messages are uniquely identifiable.
+
+In the strict sense, a _message_ is an item of data that is sent to a specific
+destination, whereas an _event_ is a signal emitted by a component upon
+reaching a given state. This document is agnostic of those differences and uses
+the term "message" in a wider sense to cover both concepts.
+
+### Producer
+
+The "producer" is a specific instance, process or device that creates and
+publishes a message. "Publishing" is the process of sending a message or batch
+to the intermediary or consumer.
```

Comment:
This should probably reference "Publishing" of the message to a destination, where the destination could either be a consumer or an intermediary.

Reply:
I tried to avoid the concept of a "destination". Events differ from messages in the sense that events aren't sent to a specific destination, but are just emitted, agnostic of who receives or acts upon the event.

Despite the misleading title, these semantic conventions aim to cover both messaging and eventing system (I tried to capture that in the definition of "message" above).

Reply:
Saw the definition of "message", but I interpreted that to refer to different people interchangeably using "message" and "event" to mean the same thing.

Is the intention for these conventions to be used for event streaming and event-driven architectures? If so, it might be worth explicitly calling that out

Reply:
Yes, I think we want to cover both event streaming and event-driven architectures. I'll put it on the agenda for the next meeting and then add clarifications according to the discussion.

Reply:
I explicitly added the terms "event streaming" and "event-driven architectures" to the motivation section.